### PR TITLE
New Game Speed: Slow Speed.

### PR DESCRIPTION
**Pull request by qqkookie**
Added new game  speed  step: Slow Game Speed: 1/4  of  normal game speed. 
And "Increase/Decrease Game Speed" shortcut key can pause the game.

### CODE REVIEW

File: data/language/ko-KR.txt [hunk 1]
```diff
@@ -4568,6 +4568,8 @@ STR_6258    :투명하게
 STR_6259    :끄기
 STR_6260    :차단된 칸 표시
 STR_6261    :넓은 보도 표시
+STR_6262    :느린 속도
+STR_6263    :매우 느린 속도
```
Comment (Broxzier) [inline data/language/ko-KR.txt]: You should translate these in the localisation repository after this gets merged instead, for PRs here we only change `en-GB.txt`.

File: src/openrct2-ui/windows/TopToolbar.cpp [hunk 2]
```diff
@@ -3056,22 +3059,24 @@ static void window_top_toolbar_tool_abort(rct_window *w, rct_widgetindex widgetI
...
 
-    gDropdownItemsArgs[0] = STR_SPEED_NORMAL;
-    gDropdownItemsArgs[1] = STR_SPEED_QUICK;
-    gDropdownItemsArgs[2] = STR_SPEED_FAST;
-    gDropdownItemsArgs[3] = STR_SPEED_TURBO;
+    gDropdownItemsArgs[0] = STR_SPEED_SLOW;
+    gDropdownItemsArgs[1] = STR_SPEED_NORMAL;
+    gDropdownItemsArgs[2] = STR_SPEED_QUICK;
+    gDropdownItemsArgs[3] = STR_SPEED_FAST;
+    gDropdownItemsArgs[4] = STR_SPEED_TURBO;
```
Comment (Broxzier) [inline src/openrct2-ui/windows/TopToolbar.cpp]: This entire block can be improved by taking the values from a static array instead, to get rid of the hardcoded numbers and string IDs.
Edit: The array already exists: `SpeedNames`

File: src/openrct2/Context.cpp [hunk 3]
```diff
@@ -809,7 +809,7 @@ namespace OpenRCT2
         bool ShouldRunVariableFrame()
         {
             if (!gConfigGeneral.uncap_fps) return false;
-            if (gGameSpeed > 4) return false;
+            if (gGameSpeed == GameSpeedHyper) return false;
```
Comment (Broxzier) [inline src/openrct2/Context.cpp]: It's safer to use `>=` instead of `==` when dealing with hard caps.

File: src/openrct2/Game.cpp [hunk 4]
```diff
@@ -127,17 +127,21 @@ GAME_COMMAND_CALLBACK_POINTER * game_command_callback_get_callback(uint32_t inde
 
 void game_increase_game_speed()
 {
-    gGameSpeed = std::min(gConfigGeneral.debugging_tools ? 5 : 4, gGameSpeed + 1);
-    if (gGameSpeed == 5)
-        gGameSpeed = 8;
+    if (gGameSpeed < (gConfigGeneral.debugging_tools ? GameSpeedHyper : GameSpeedTurbo))
+        gGameSpeed++;
+
+    if ( game_is_paused() )
```
Comment (Broxzier) [inline src/openrct2/Game.cpp]: no spaced inside the parentheses. Use clang-format to format correctly automatically.

File: src/openrct2/Game.h [hunk 5]
```diff
@@ -138,6 +138,7 @@ extern uint32_t gCurrentTicks;
 extern uint16_t gTicksSinceLastUpdate;
 extern uint8_t  gGamePaused;
 extern int32_t gGameSpeed;
+enum { GameSpeedPaused = 0, GameSpeedNormal = 2, GameSpeedTurbo = 5, GameSpeedHyper = 6 };
```
Comment (Broxzier) [inline src/openrct2/Game.h]: This shouldn't all be on one line, and moved up with the other enums.

File: src/openrct2/GameState.cpp [hunk 6]
```diff
@@ -73,7 +73,10 @@ void GameState::Update()
 {
     gInUpdateCode = true;
 
-    uint32_t numUpdates;
+    uint32_t numUpdates = 0;
+
+    static uint32_t pctUpdates = 0;
+    static uint32_t GameSpeeds[] = { 0, 25, 100, 200, 400, 1000, 10000 };  // speed in percent
```
Comment (Broxzier) [inline src/openrct2/GameState.cpp]: Move the comment to be before this line, and add a trailing comma please.

**Comment by Broxzier** [comment]
I just tested it out, and it works, though there are three things:

 1. The normal speed button now shows two >> instead of one. If you could add a new image for half speeds.
 2. I can't find the 1/4 speed setting.
 3. It looks really choppy (as expected) but with 'uncap FPS' on it should interpolate between frames properly. Now it looks like it interpolates with the default game speed. This may be out of scope for this PR, though.

Don't forget to add yourself to `contributors.md`, and add an entry to the changelog.

File: src/openrct2/Game.h (hunk 5 above)
Comment (IntelOrca) [reply src/openrct2/Game.h]: Constants and enums should be upper case.

File: data/language/ko-KR.txt [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# STR_XXXX part is read and XXXX becomes the string id number.
+﻿# STR_XXXX part is read and XXXX becomes the string id number.
```
Comment (Broxzier) [inline data/language/ko-KR.txt]: You accidentally added the BOM here. Some editors automatically add this with UTF-8 encoded files.

File: distribution/changelog.txt [hunk 8]
```diff
@@ -1,10 +1,11 @@
-0.2.0+ (in development)
+﻿0.2.0+ (in development)
```
Comment (Broxzier) [inline distribution/changelog.txt]: Same here. Try using another editor, like Notepad++.

File: src/openrct2-ui/title/TitleSequencePlayer.cpp [hunk 9]
```diff
@@ -267,8 +267,54 @@ class TitleSequencePlayer final : public ITitleSequencePlayer
                     return false;
                 }
 
+<<<<<<< HEAD
```
Comment (Broxzier) [inline src/openrct2-ui/title/TitleSequencePlayer.cpp]: This merge must have slipped in.

File: resources/g2/sprites.json [hunk 10]
```diff
@@ -1223,5 +1223,5 @@
         "y_offset": 4,
         "palette": "keep",
         "forceBmp": true
-    }
+    },
```
Comment (Broxzier) [inline resources/g2/sprites.json]: The travis builds fail because of this new comma. I think you meant to add the new image here.

**Comment by Broxzier** [comment]
It looks like you're having some trouble with git, would you like some help with cleaning this up?

**Pull request closed without merging**